packages/runtime/src/state/modules/read-only-documents.ts

```diff
@@ -20,24 +20,17 @@ export function createDocumentReference(key: string): DocumentReference {
 export type Document = {
   key: string
   rootElement: Element
+  locale: string | null
```

Comment:
Unfortunately, this `Document` type is used in some actions as the arg and the payload, for example, the `registerDocument` and the `registerBuilderDocument`:
```ts
type RegisterDocumentAction = {
  type: typeof ActionTypes.REGISTER_DOCUMENT
  payload: { documentKey: string; document: Document }
}
```
Perhaps we need to decouple the type of the state and the actions, so that we don't accidentally change the payload of the action when we update the state. Otherwise it could cause an error because the builder is expecting a payload that's different on an old runtime.

Reply:
Decoupled `Document` from document payload 🙏.